server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java

```diff
@@ -48,25 +51,27 @@ public Extent getExtent() throws IOException {
     /**
      * Returns true if the rectangle query and the edge tree's shape overlap
      */
-    public boolean intersects(Extent extent) throws IOException {
-        if (hasArea) {
-            return containsBottomLeft(extent) || crosses(extent);
-        } else {
-            return crosses(extent);
+    @Override
+    public GeoRelation relate(Extent extent) throws IOException {
```

Comment:
I think we can move the check of the extent here. Not needed for this PR but I think it is worthy to note.

Reply:
I agree. consolidating to one relating method reduces the places that there is an extent check. I will follow-up